client/src/components/User/UserProfile.js

```diff
@@ -1,13 +1,143 @@
-import React, { Fragment } from "react";
+import React, { useState } from "react";
 import { useSelector, useDispatch } from "react-redux";
+import "./UserProfile.css";
+import CustomButton from "../Custom/CustomButton";
+import CustomFormInput from "../Custom/CustomFormInput";
+import { useNavigate } from "react-router-dom";
+import { useForm } from "react-hook-form";
+import { auth } from "../../firebase/firebaseConfig";
+import { updateProfile } from "firebase/auth";
+import { updateUserInfo } from "../../redux/userSlice.js";
 
 function UserProfile() {
-  const user = useSelector((state) => state.user);
-  console.log(user);
+  const user = useSelector((state) => state.user.user);
+  const navigate = useNavigate();
+  const dispatch = useDispatch();
+  const { handleSubmit, reset, register } = useForm({});
```

Comment:
is the empty object needed as a parameter for useForm? if not, please remove it, thanks!

Reply:
done